### PR TITLE
Add the cards for previewing the careful explanations when they appear in addition to other resources

### LINKED ISSUE #730: Implement the carefully worded description cards in the learn page
Implementation: 
Provide 2 dedicated URLs for carefully worded beginners.

Make the short version to open a full version in a new window.

### PR DESCRIPTION
**Pull request by delapuente**
Fix #730

This PR does, mainly, 3 things:
- It factors out the explanations, introducing a `compact` mode and a summary slot for displaying it alone when `compact` is present. That view becomes the card.
- It introduces URL-based filters so the different filter combinations have now a unique URL. For doing it:
  1. The `beforeRouteEnter` and `beforeRouteUpdate` functions parse the URL for the `learnLevel` and the `timeScale` parameters and set them in the store (both functions are needed to be aware of query parameters changes).
  2. Filters no longer change the store, they now navigate to a new URL (this navigation is virtual, using the programmatic version of `<nuxt-link>` which is `this.$router.push()`).
  3. Everything else remains the same.

Regarding the visibility of the explanations:
- The explanations are shown:
  - In compact more when accompanied by more resources.
  - Extended when alone.
  - Not displayed at all when showing **everything**.
- It removes the 1 minute/advance classification for one of the resources, these 1 minute/whatever combinations are restricted to the texts.

Although, it also includes these somehow related fixes:
- As part of #724, makes all the texts `cool-gray-80` to match Figma designs.
- It also distinguishes the new kind of text introduced by the learning page: the content paragraph. This type of paragraph has become the standard paragraph. Outstanding paragraphs are now marked by entity modifiers `_importance_outstanding` while the previous old standard paragraph, which acted as a support text for sections (in contras to content) is now `_importance_support`.

And this unrelated one, but it hurt me seeing it there:
- Fix a bug regarding repeated keys when generating the second-level links of the menu.

### CODE REVIEW

File: pages/learn.vue [hunk 1]
```diff
@@ -149,23 +169,74 @@ export default class extends QiskitPage {
...
+        ...this.$route.query,
+        [paramName]: value
+      }
+    })
   }
 
-  get isAMinuteForBeginner (): boolean {
-    return (this as any).timeScale === TIME_SCALES.minute &&
-      (this as any).learnLevel === LEARN_LEVELS.beginner
+  get showingEverything (): boolean {
```
Comment (y4izus) [inline pages/learn.vue:214]: Per offline conversations, I agree that `isShowingEverything` is easy to understand than `showingEverything`. Same for the rest of the checks

File: pages/learn.vue [hunk 2]
```diff
@@ -149,23 +169,74 @@ export default class extends QiskitPage {
...
+  get showingOneMinuteForAdvanced (): boolean {
+    return this.showingOneMinute && this.showingLevel(LEARN_LEVELS.advanced)
+  }
+
+  get showingOneMinute (): boolean {
+    const { timeScale } = (this as any)
+    return [TIME_SCALES.any, TIME_SCALES.minute].includes(timeScale)
+  }
+
+  showingLevel (level: LearnLevel): boolean {
```
Comment (y4izus) [inline pages/learn.vue:232]: Are you missing the `get` here?

File: components/learn/TheCarefulExplanationForAdvanced.vue [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
-    <p class="copy__paragraph">
-      There are many misconceptions about quantum computers, this is understandable since the science is exciting and potentially very useful, but is unfortunately not easily accessible to someone without the appropriate background. Part of the problem is that some quantum mechanical concepts cannot be analogised to everyday experiences, so that has been avoided here.
-    </p>
+  <CarefulExplanation v-bind="$attrs">
+    <template #summary>
+      <h2 class="copy__subtitle">
+        A Carefully Worded Summary of Quantum Computing for Advanced
+      </h2>
+      <p class="copy__paragraph">
+        There are many misconceptions about quantum computers, this is understandable since the science is exciting and potentially very useful, but is unfortunately not easily accessible to someone without the appropriate background. Part of the problem is that some quantum mechanical concepts cannot be analogised to everyday experiences, so that has been avoided here.
```
[the remark this answers is not in the document]
Comment (delapuente) [reply components/learn/TheCarefulExplanationForAdvanced.vue:8]: I will wait for Leron for the copies. They are not still final.

File: pages/learn.vue (hunk 2 above)
Comment (delapuente) [reply pages/learn.vue:232]: No, all others are properties but this one is a function that recieves the learn level you're checking.

File: components/learn/TheCarefulExplanationForAdvanced.vue [hunk 4]
```diff
@@ -1,11 +1,13 @@
 <template>
```
[the remark this answers is not in the document]
Comment (delapuente) [reply components/learn/TheCarefulExplanationForAdvanced.vue:1]: I thought about it, but it is harder to make someone to update a custom JSON rather than HTML. However, perhaps we can leverage content/nuxt for doing this during importation. Let's see later.

File: new-content/learning-resources/textbook.md [hunk 5]
```diff
@@ -4,9 +4,9 @@ to: /textbook
 image: /images/landing-page/feature-hardware.jpg
```
Comment (y4izus) [inline new-content/learning-resources/textbook.md:4]: This comment is for the previous line: `to: /textbook`

If we use as URL `/textbook` instead of `https://qiskit.org/texbook`, the `AppLink` component classifies it as "no external" link and converts it into a `nuxt-link`. 
Screen Shot 2020-07-16 at 10 29 34

When we click on it, we get an error.
Screen Shot 2020-07-16 at 10 30 09

File: pages/learn.vue [hunk 6]
```diff
@@ -82,11 +87,22 @@
...
           <section class="the-learning-resources-list__results">
-            <TheCarefulExplanationForBeginner v-if="isAMinuteForBeginner" />
-            <TheCarefulExplanationForAdvanced v-if="isAMinuteForAdvanced" />
+            <TheCarefulExplanationForBeginner
+              v-if="isShowingOneMinuteFor(learnLevels.beginner) && !isShowingEverything"
+              class="the-learning-resources-list__item"
+              :compact="isShowingMoreResources"
+              url="/learn/?learnLevel=beginner&amp;timeScale=1%20minute"
+            />
+            <TheCarefulExplanationForExperts
```
Comment (y4izus) [inline pages/learn.vue:96]: Just for curiosity: why did you change "Advanced" for "Expert" in some parts of the code?

File: pages/learn.vue [hunk 7]
```diff
@@ -107,20 +123,23 @@ import { Component } from 'vue-property-decorator'
...
+  TIME_SCALE_OPTIONS,
+  LearnLevel
 } from '~/store/modules/learning-resources.ts'
 
 @Component({
   components: {
     LearningResourceCard,
-    TheCarefulExplanationForAdvanced,
-    TheCarefulExplanationForBeginner
+    TheCarefulExplanationForExperts,
```
Comment (y4izus) [inline pages/learn.vue:140]: We are using "Experts" and "Beginner", would you mind to add an "s" to "beginner" to follow the same structure?

File: store/modules/learning-resources.ts [hunk 8]
```diff
@@ -10,8 +10,8 @@ const TIME_SCALES = Object.freeze({
```
Comment (y4izus) [inline store/modules/learning-resources.ts:10]: This is a comment for the line: `any: 'any'`

`any` is a common word used on TS. I think we should replace it with `all` (just for the variable name, not for the label). What do you think?

File: components/ui/AppCta.vue [hunk 9]
```diff
@@ -1,10 +1,11 @@
 <template>
   <AppLink
+    ref="link"
     class="app-cta copy__link"
     v-bind="$attrs"
   >
     <div><slot /></div>
-    <ArrowRight16 />
+    <component :is="isExternal ? 'Launch16' : 'ArrowRight16'" />
```
Comment (y4izus) [inline components/ui/AppCta.vue:8]: Really good idea to use the "launch" icon for external links! :)

File: new-content/learning-resources/textbook.md (hunk 5 above)
Comment (delapuente) [reply new-content/learning-resources/textbook.md:4]: Second error is normal, the textbook is not mounted during development time.

Regarding the one, I'm fine adding the complete URL to force it to be open is a new tab.

File: pages/learn.vue (hunk 6 above)
Comment (delapuente) [reply pages/learn.vue:96]: Because of the title of the text, to make it match with:
"A Carefully Worded Summary of Quantum Computing for Experts"

File: new-content/learning-resources/textbook.md (hunk 5 above)
Comment (delapuente) [reply new-content/learning-resources/textbook.md:4]: I'll fix this when reviewing #749
